src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -656,6 +658,23 @@ public <T> T[] toArray(T[] a) {
             }
             return array;
         }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public void forEach(Consumer<? super E> action) {
+            action.accept(e0); // implicit null check
+            if (e1 != EMPTY) {
+                action.accept((E) e1);
+            }
+        }
+
+        @Override
+        public Spliterator<E> spliterator() {
+            if (e1 == EMPTY) {
+                return Collections.singletonSpliterator(e0);
+            }
+            return super.spliterator();
```

Comment:
Not a big deal, but I prefer using if/else for situations like this where the then-part and else-part are equal in size and are conceptually at the same level. (As opposed to a check for a quick special case, where an early return is sensible, followed by the main logic at the method's top level.) Code in the rest of the file is similar, I think.

Reply:
Done. Thanks!